worlds/metroidprime/__init__.py

```diff
@@ -0,0 +1,473 @@
+    get_world_door_mapping,
+    remap_doors_to_power_beam_if_necessary,
+)
+
+from worlds.LauncherComponents import Component, SuffixIdentifier, Type, components, launch_subprocess  # type: ignore
+import settings
+from worlds.AutoWorld import World, WebWorld
+from .data.Transports import (
+    ELEVATOR_USEFUL_NAMES,
+    default_elevator_mappings,
+    get_random_elevator_mapping,
+)
+from .Config import make_config
+from .Regions import create_regions
+from .Locations import every_location
+from .ItemPool import generate_item_pool, generate_base_start_inventory
+from .PrimeOptions import (
+    BlastShieldRandomization,
+    DoorColorRandomization,
+    MetroidPrimeOptions,
+    prime_option_groups,
+)
+from .Items import (
+    MetroidPrimeItem,
+    SuitUpgrade,
+    artifact_table,
+    item_table,
+)
+from .data.StartRoomData import (
+    StartRoomData,
+    init_starting_beam,
+    init_starting_loadout,
+    init_starting_room_data,
+)
+from .Container import MetroidPrimeContainer
+from BaseClasses import MultiWorld, Tutorial, ItemClassification
+
+
+class MultiworldWithPassthrough(MultiWorld):
+    re_gen_passthrough: Dict[str, Dict[str, Any]] = {}
+
+
+def run_client(*args: Any):
+    from .MetroidPrimeClient import launch
+
+    launch_subprocess(launch, name="MetroidPrimeClient")
+
+
+components.append(
+    Component(
+        "Metroid Prime Client",
+        func=run_client,
+        component_type=Type.CLIENT,
+        file_identifier=SuffixIdentifier(".apmp1"),
+    )
+)
+
+
+class MetroidPrimeSettings(settings.Group):
+    class RomFile(settings.UserFilePath):
+        """File name of the Metroid Prime ISO"""
+
+        description = "Metroid Prime GC ISO file"
+        copy_to = "Metroid_Prime.iso"
+
+    class RomStart(str):
+        """
+        Set this to false to never autostart a rom (such as after patching),
+        Set it to true to have the operating system default program open the iso
+        Alternatively, set it to a path to a program to open the .iso file with (like Dolplhin)
+        """
+
+    rom_file: RomFile = RomFile(RomFile.copy_to)
+    rom_start: typing.Union[RomStart, bool] = False
+
+
+class MetroidPrimeWeb(WebWorld):
+    tutorials = [
+        Tutorial(
+            "Multiworld Setup Guide",
+            "A guide to setting up Metroid Prime for Archipelago",
+            "English",
+            "setup.md",
+            "setup/en",
+            ["hesto2", "Electro15"],
+        )
+    ]
+    option_groups = prime_option_groups
+
+
+class MetroidPrimeWorld(World):
+    """
+    Metroid Prime is a first-person action-adventure game originally for the Gamecube. Play as
+    the bounty hunter Samus Aran as she traverses the planet Tallon IV and uncovers the plans
+    of the Space Pirates.
+    """
+
+    game = "Metroid Prime"
+    web = MetroidPrimeWeb()
+    required_client_version = (0, 5, 0)
+    options_dataclass = MetroidPrimeOptions
+    options: MetroidPrimeOptions  # type: ignore

```

Comment:
I'm not sure the reason for this `# type: ignore`, this should be fine.
```suggestion
    options: MetroidPrimeOptions
```